akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/FutureDirectivesSpec.scala

```diff
@@ -80,18 +80,25 @@ class FutureDirectivesSpec extends RoutingSpec with Inside {
     }
     "fail fast if the circuit breaker is open" in new TestWithCircuitBreaker {
       openBreaker()
-      Get() ~> onCompleteWithBreaker(breaker)(Future.successful(1)) { echoComplete } ~> check {
-        inside(rejection) {
-          case CircuitBreakerOpenRejection(_) ⇒
-        }
-      }
+      // since this is timing sensitive, try a few times to observe the breaker open
+      awaitAssert(
+        Get() ~> onCompleteWithBreaker(breaker)(Future.successful(1)) { echoComplete } ~> check {
+          inside(rejection) {
+            case CircuitBreakerOpenRejection(_) ⇒
+          }
+        }, 500.millis)
     }
     "stop failing fast when the circuit breaker closes" in new TestWithCircuitBreaker {
       openBreaker()
-      Thread.sleep(breakerResetTimeout.toMillis + 200)
-      Get() ~> onCompleteWithBreaker(breaker)(Future.successful(1)) { echoComplete } ~> check {
-        responseAs[String] shouldEqual "Success(1)"
-      }
+      // observe that it opened
+      awaitAssert(breaker.isOpen should ===(true))
+      Thread.sleep(breakerResetTimeout.toMillis)
+      // since this is timing sensitive, try a few times to observe the breaker closed
+      awaitAssert({
+        Get() ~> onCompleteWithBreaker(breaker)(Future.successful(1)) { echoComplete } ~> check {
+          responseAs[String] shouldEqual "Success(1)"
+        }
+      }, 500.millis)
```

Comment:
Do we even need the sleep still? We could just succeed once the breaker is reset by increasing the timeout here to `breakerResetTimeout` plus a bit.

Reply:
Yeah, I just thought it wasn't very useful to hit it while we know it will fail, either way is fine, I'll change.

Reply:
One thing that may be a bit surprising with that is that the `breakerResetTimeout` is already dilated, and then `assertAwait` further dilates the max time.